final/design.tex

```diff
@@ -1,6 +1,49 @@
 \chapter{Design}
 \label{cha:design}
 
+As detailed in the previous chapters, the Spoofax language workbench already
+offers a wide variety of services to assist in developing new domain specific
+language. However, not all exposed functionality can be easily used from within
+the context of a REPL, where a user just wants to be able to type in some text
+and see a result.
+
+An example is the various transformation goals made available to the user by a
+language designer. There is no such thing as a uniform ``evaluation command'',
+that works across all languages, but instead each language defines its own
+transform goals, of which one is an evaluation goal.  Furthermore the sequence
+of processing steps needed to go from source to parsed source and transformed
+result also varies between languages. One of the main goals the design had to
+accomplish was therefore to expose a uniform interface for the various
+transformation stages to REPL clients.
+
+In \cref{sec:overview} an overview of the product design and its various
+components will be given first. Afterwards the individual components will be
```

Comment:
We decided we would not use future tense for this, so: `an overview of the product design and its various components is given first`. Et cetera...